src/thunderbots/software/ai/navigator/obstacle/obstacle.cpp

```diff
@@ -0,0 +1,109 @@
+
+#include "obstacle.h"
+
+const Polygon& Obstacle::getBoundaryPolygon() const
+{
+    return _polygon;
+}
+
+Obstacle::Obstacle() : _polygon{Point(0, 0), Point(0, 1), Point(1, 1), Point(1, 0)} {}
+
+Obstacle::Obstacle(Polygon polygon) : _polygon(polygon) {}
+
+Obstacle Obstacle::createPlaceholderObstacle()
+{
+    return Obstacle();
+}
+
+Obstacle Obstacle::createRobotObstacle(const Robot& robot, bool enable_velocity_cushion)
+{
+    // TODO: figure out how to get tick length
+    double tick_length = 1.0;
+
+    // TODO: figure out actual factor (not 2) needed to clear two robot based on hexagon
+    // implementation
+    double radius_cushion = ROBOT_MAX_RADIUS_METERS * 2;
+
+    // vector in the direction of the velocity and with the scaled size of the velocity
+    Vector velocity_cushion_vector =
+        robot.velocity().norm(robot.velocity().len() * tick_length);
+
+    return createRobotObstacleFromPositionAndRadiusAndVelocity(
+        robot.position(), radius_cushion, velocity_cushion_vector,
+        enable_velocity_cushion);
+}
+
+Obstacle Obstacle::createRobotObstacleWithScalingParams(
+    const Robot& robot, bool enable_velocity_cushion, double radius_cushion_scaling,
+    double tick_length, double velocity_cushion_scaling)
+{
+    double radius_cushion = ROBOT_MAX_RADIUS_METERS * 4 / std::sqrt(3) * radius_cushion_scaling;
+
+    // vector in the direction of the velocity and with the scaled size of the velocity
+    Vector velocity_cushion_vector = robot.velocity().norm(
+        robot.velocity().len() * tick_length * velocity_cushion_scaling);
+
+    return createRobotObstacleFromPositionAndRadiusAndVelocity(
+        robot.position(), radius_cushion, velocity_cushion_vector,
+        enable_velocity_cushion && velocity_cushion_vector.len() > radius_cushion);
```

Comment:
Based on my knowledge of ticks and robot speed, I don't think this will ever evaluate to true. Ticks are usually very small. The data we get from the vision system has a tick rate of 1/30.

Reply:
Physically, this is supposed to mean that if the projected distance travelled in one tick (including buffers/scaling) is greater a robot diameter (including buffer/scaling), then we need to create a rectangle in front of that robot to avoid.

Reply:
I still don't fully agree scaling things for these checks should be based off the tick speed. IMO we should be pretty agnostic of the tick speed of the application.

My main concerns would be
* Since the tick speed is unregulated, there could be very high variance in the values. If we are scaling by large values to make this comparable to other values (for the conditional checks), we could have very high variance in those values, and have wildly varying obstacles (flickering between largely projected and not)
* I'm pretty sure I understand what you're trying to accomplish with this based on what you told me in person, but I think it's an over-optimization now, and potentially tricky based on my above point. I generally think just using the robot's velocity is better since it's more consistent and stable.

Reply:
OK yeah I see. It's redundant because we can already modify the size of the velocity cushion, so I'll remove it.